pkgs/development/python-modules/bugwarrior/default.nix

```diff
@@ -1,68 +1,122 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
-  pythonOlder,
+  fetchFromGitHub,
+  taskwarrior3,
+  versionCheckHook,
+  writableTmpDirAsHomeHook,
   setuptools,
-  twiggy,
-  requests,
-  offtrac,
-  python-bugzilla,
-  taskw,
-  python-dateutil,
-  pytz,
-  keyring,
-  six,
-  jinja2,
-  pycurl,
+  click,
   dogpile-cache,
+  importlib-metadata,
+  jinja2,
   lockfile,
-  click,
-  pyxdg,
-  future,
+  pydantic,
+  python-dateutil,
+  pytz,
+  requests,
+  taskw,
+  debianbts,
+  python-bugzilla,
+  google-api-python-client,
+  google-auth-oauthlib,
   jira,
+  keyring,
+  offtrac,
+  pytestCheckHook,
+  docutils,
+  pytest-subtests,
+  responses,
+  ruff,
+  sphinx,
+  sphinx-click,
+  sphinx-inline-tabs,
 }:
 
 buildPythonPackage rec {
   pname = "bugwarrior";
-  version = "1.8.0";
-  format = "setuptools";
-  disabled = pythonOlder "3.6";
+  version = "2.0.0";
+  pyproject = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "f024c29d2089b826f05481cace33a62aa984f33e98d226f6e41897e6f11b3f51";
+  src = fetchFromGitHub {
+    owner = "GothenburgBitFactory";
+    repo = "bugwarrior";
+    tag = version;
+    hash = "sha256-VuHTrkxLZmQOxyig2krVU9UZDDbLY08MfB9si08lh3E=";
   };
 
-  propagatedBuildInputs = [
-    setuptools
-    twiggy
-    requests
-    offtrac
-    python-bugzilla
-    taskw
-    python-dateutil
-    pytz
-    keyring
-    six
-    jinja2
-    pycurl
+  build-system = [ setuptools ];
+
+  dependencies = [
+    click
     dogpile-cache
+    importlib-metadata
+    jinja2
     lockfile
-    click
-    pyxdg
-    future
-    jira
+    pydantic
+    python-dateutil
+    pytz
+    requests
+    taskw
+  ]
+  ++ pydantic.optional-dependencies.email;
+  pythonRemoveDeps = [ "tomli" ];
+
+  optional-dependencies = {
+    bts = [ debianbts ];
+    bugzilla = [ python-bugzilla ];
+    gmail = [
+      google-api-python-client
+      google-auth-oauthlib
+    ];
+    jira = [ jira ];
+    keyring = [ keyring ];
+    trac = [ offtrac ];
+  };
+
+  nativeCheckInputs = [
+    taskwarrior3
+    versionCheckHook
+    writableTmpDirAsHomeHook
+  ]
+  ++ [
+    pytestCheckHook
+    docutils
+    pytest-subtests
+    responses
+    ruff
+    sphinx
+    sphinx-click
+    sphinx-inline-tabs
```

Comment:
These are required for tests?

Reply:
Yes, there are tests of the docs build. Those tests aren't at all helpful as a diagnostic for nixpkgs but to me it seems a little cleaner to include the dependencies than to disable them since they're already packaged.